### PR TITLE
Verify error stack is empty after each operation when using Reference…

### PR DESCRIPTION
**Pull request by normanmaurer**
…CountedOpenSslEngine.

Motivation:

https://github.com/netty/netty/pull/7941 proved that its easy to not correctly clear the error stack sometimes. We should do carefully test this.

Modifications:

Add a new SSLEngine wrapper that is used during tests, which verifies that the error stack is empty after each method call.

Result:

Better testing.

### CODE REVIEW

**Comment by normanmaurer** [comment]
As a followup for https://github.com/netty/netty/pull/7941

File: handler/src/test/java/io/netty/handler/ssl/OpenSslErrorStackAssertSSLEngine.java [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+    }
+
+    @Override
+    public boolean release() {
+        return getWrappedEngine().release();
+    }
+
+    @Override
+    public boolean release(int decrement) {
+        return getWrappedEngine().release();
```
Comment (carl-mastrangelo) [inline handler/src/test/java/io/netty/handler/ssl/OpenSslErrorStackAssertSSLEngine.java]: missing param?

Comment (normanmaurer) [reply handler/src/test/java/io/netty/handler/ssl/OpenSslErrorStackAssertSSLEngine.java]: ups... will fix it quickly.

Comment (normanmaurer) [reply handler/src/test/java/io/netty/handler/ssl/OpenSslErrorStackAssertSSLEngine.java]: https://github.com/netty/netty/commit/47985c11c1d9f11b94018539108ed65f0c6a32ab
